### PR TITLE
Fixes #18532 - Better help for unsupported hammer commands

### PR DESCRIPTION
**Pull request by ofedoren**

### CODE REVIEW

**Comment by tstrachota** [comment]
[test]

File: lib/hammer_cli/apipie/command.rb [hunk 1]
```diff
@@ -36,6 +36,14 @@ def execute
       return HammerCLI::EX_OK
     end
 
+    def help
+      help_str = super
+      if not resource or (not action.nil? and not resource.has_action?(action))
```
[the remark this answers is not in the document]
Comment (mbacovsky) [reply lib/hammer_cli/apipie/command.rb]: I also found https://github.com/theforeman/hammer-cli/blob/master/lib/hammer_cli/apipie/resource.rb#L55 where we do similar check. Might be nice if we are consistent on how we indicate Command and API incompatibility.

@tstrachota, would it be acceptable if we `warn` the massage (with details such as resource and action going to `logger.warn` perhaps) in missing resource and action? This way the message would be displayed also in help, at the top though.

[the remark this answers is not in the document]
Comment (tstrachota) [reply lib/hammer_cli/apipie/command.rb]: I'd prefer if the warning text was part of the help output as I suggest in the redmine issue.

Hammer commands should check if their resource exist and end gracefully when their resource isn't found in the api docs. But it should be a separate fix. Currently they most likely fail somewhere in the middle of the process due to a method being called on nil.

**Comment by ofedoren** [comment]
@mbacovsky I've rephrased as you suggested. Also I think that your rephrased condition checks for a little bit different thing. I was patterned on  https://github.com/ofedoren/hammer-cli/blob/29a4335fc88ac77a2aae13adf12ff9d8c67ad8ce/lib/hammer_cli/apipie/command.rb#L52-L54

**Comment by tstrachota** [comment]
@mbacovsky ping. I guess that this is waiting for your response.

**Comment by mbacovsky** [comment]
[test]

**Comment by mbacovsky** [comment]
:+1: @ofedoren, thanks for the update. Looks good now.

**Pull request merged**